pages/sign_in.tsx

```diff
@@ -1,134 +1,212 @@
-import { LinkIcon } from "@heroicons/react/solid";
+import useStorage from "hooks/useStorage";
 import { useTranslation } from "next-i18next";
 import { serverSideTranslations } from "next-i18next/serverSideTranslations";
-import Link from "next/link";
 import { useRouter } from "next/router";
-import { ChangeEvent, ReactElement, useState } from "react";
-import BrInput from "../components/brickroom/BrInput";
-import KeyringGeneration from "../components/KeyringGeneration";
-import NRULayout from "../components/layout/NRULayout";
-import VerifySeed from "../components/VerifySeed";
+import { ReactElement, useEffect, useState } from "react";
+import * as yup from "yup";
 import { useAuth } from "../hooks/useAuth";
-import devLog from "../lib/devLog";
-import { NextPageWithLayout } from "./_app";
+import type { NextPageWithLayout } from "./_app";
+
+// Login functions
+import keypairoomClientRecreateKeys from "zenflows-crypto/src/keypairoomClientRecreateKeys";
+import { zencode_exec } from "zenroom";
+
+// Layout
+import NRULayout from "../components/layout/NRULayout";
+
+// Partials
+import Passphrase from "components/partials/auth/Passphrase";
+import Questions, { QuestionsNS } from "components/partials/auth/Questions";
+import ChooseMode from "components/partials/sign_in/ChooseMode";
+import EnterEmail, { EnterEmailNS } from "components/partials/sign_in/EnterEmail";
+import ViaPassphrase, { ViaPassphraseNS } from "components/partials/sign_in/ViaPassphrase";
+import ViaQuestions from "components/partials/sign_in/ViaQuestions";
+
+// Components
+import BrAuthSuggestion from "components/brickroom/BrAuthSuggestion";
+import BrError from "components/brickroom/BrError";
+
+//
 
 export async function getStaticProps({ locale }: any) {
   return {
     props: {
-      ...(await serverSideTranslations(locale, ["signInProps", "signUpProps"])),
+      ...(await serverSideTranslations(locale, ["signInProps", "authProps"])),
     },
   };
 }
 
+//
+
 const Sign_in: NextPageWithLayout = () => {
+  const { t } = useTranslation("signInProps");
+  const { register, login } = useAuth();
+  const { getItem, setItem } = useStorage();
+  const router = useRouter();
+
+  //
+
   const [isPassprhase, setIsPassphrase] = useState(false);
+  const [isQuestions, setIsQuestions] = useState(false);
   const [step, setStep] = useState(0);
-  const [email, setEmail] = useState("");
-  const [pdfk, setPdfk] = useState("");
-  const [isMailExisting, setIsMailExising] = useState(true);
+  const [error, setError] = useState("");
 
-  const { register, login } = useAuth();
+  const [signInData, setSignInData] = useState({
+    email: "",
+    pdfk: "",
+    seed: "",
+  });
+
+  //
+
+  const emailEntered = async (data: EnterEmailNS.FormValues) => {
+    // Getting HMAC
+    const result = await register(data.email, false);
+    const key = result.keypairoomServer;
+    //
+    console.log(key);
```

Comment:
please beware this logs... maybe to help you can configure eslint with https://eslint.org/docs/latest/rules/no-console

Reply:
I'll use `devLog` made by ennio next time. But yeah, maybe we can also enforce it